### PR TITLE
postgres metrics

### PR DESCRIPTION
**Pull request by QuarpT**
postgres metrics for cloudformation.
Will help us find expensive queries

### CODE REVIEW

File: app/models/database/postgres/PostgresDeletedUserRepository.scala [hunk 1]
```diff
@@ -10,29 +14,33 @@ import scalikejdbc._
...
-           | WHERE jdoc@>$idMatcher::jsonb
-           | OR jdoc@>$emailMatcher::jsonb
-           | OR jdoc@>$usernameMatcher::jsonb
-           | order by jdoc->>'username'
+  implicit val ec = ExecutionContext.fromExecutor(Executors.newCachedThreadPool())
+
+  def findBy(query: String): ApiResponse[Option[DeletedUser]] = withMetricsFE("deletedUser.findBy", query) {
+    readOnly { implicit session =>
+      val _query = query.toLowerCase
+      val idMatcher = s"""{"_id":"${_query}"}"""
```
Comment (rahilb) [inline app/models/database/postgres/PostgresDeletedUserRepository.scala]: I know I wrote this, but I think we should change these to be constructed via the json AST then compactRendered or whatever so the string is always valid json.

Comment (rahilb) [reply app/models/database/postgres/PostgresDeletedUserRepository.scala]: alternatively we can use `#>>` instead and we can then just use a string parameter.

File: app/models/database/postgres/PostgresUserRepository.scala [hunk 2]
```diff
@@ -15,14 +17,16 @@ import scalaz.-\/
...
-  with PostgresUtils {
+  with PostgresUtils
+  with MetricsSupport{
 
   implicit val ec = ExecutionContext.fromExecutor(Executors.newCachedThreadPool())
 
-  def search(query: String, limit: Option[Int] = None, offset: Option[Int] = None): ApiResponse[SearchResponse] = {
-    val _offset = offset.getOrElse(0)
+  def search(query: String, limit: Option[Int] = None, offset: Option[Int] = None): ApiResponse[SearchResponse] = withMetricsFE("user.search", s"$query $limit $offset") {
+    val _offset = math.min(offset.getOrElse(0), SearchValidation.maxOffset)
```
Comment (rahilb) [inline app/models/database/postgres/PostgresUserRepository.scala]: there is already a max offset that is enforced at UserServlet line 46

File: app/actors/metrics/MetricsSupport.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+import scalaz.{-\/, EitherT, \/, \/-}
+
+object MetricsSupport {
+  case class Namespace(name: String)
+}
+
+trait MetricsSupport extends LazyLogging {
```
Comment (rahilb) [inline app/actors/metrics/MetricsSupport.scala]: It's a shame we didn't choose to use some well known metrics library and opted to roll our own, resulting in code duplication like this.

Comment (QuarpT) [reply app/actors/metrics/MetricsSupport.scala]: Controversial question... do you think it's worth importing a library for something small given the dependency hell it introduces? 

Frameworks & SDKs good, small libraries bad 😆

Comment (rahilb) [reply app/actors/metrics/MetricsSupport.scala]: I don't think it's small, there's actually a lot that is easy to get wrong with metrics ime

Comment (QuarpT) [reply app/actors/metrics/MetricsSupport.scala]: The amazon cloudwatch library is the underlying metrics library. This is a thin wrapper around it, but I understand what you're saying!

File: app/util/StopWatch.scala [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package util
+
+import java.time.Instant
+
+class StopWatch {
+  private val startTime = Instant.now().toEpochMilli
```
Comment (rahilb) [inline app/util/StopWatch.scala]: private[this] can be used here.

File: app/models/database/postgres/PostgresUserRepository.scala (hunk 2 above)
Comment (QuarpT) [reply app/models/database/postgres/PostgresUserRepository.scala]: 👍

**Pull request merged**